File: 3532. Path Existence Queries in a Graph I.cpp
class Solution {
public:
    vector<bool> pathExistenceQueries(int n, vector<int>& nums, int maxDiff, vector<vector<int>>& queries) {
        vector<int> path(n,0);

        for(int i=1;i<n;i++)
        {
            int isNew=nums[i]-nums[i-1]>maxDiff;

            path[i]=path[i-1]+isNew;
        }
        vector<bool> ans;
        for(int i=0;i<queries.size();i++)
        {
            int u=queries[i][0];
            int v=queries[i][1];

            ans.push_back(path[u]==path[v]);
        }
        return ans;
    }
};

File: 3532. Path Existence Queries in a Graph I.py
from typing import List
class Solution:
    def pathExistenceQueries(self, n: int, nums: List[int], maxDiff: int, queries: List[List[int]]) -> List[bool]:
        path = [0] * n

        for i in range(1, n):
            is_new = 1 if nums[i] - nums[i - 1] > maxDiff else 0
            path[i] = path[i - 1] + is_new

        ans = []

        for u, v in queries:
            ans.append(path[u] == path[v])

        return ans